base-image/entrypoint.py

```diff
@@ -144,16 +187,14 @@ def perform_processing(transformer_instance, args: argparse.ArgumentParser, meta
         if hasattr(transformer_instance, 'get_transformer_params'):
             transformer_params = transformer_instance.get_transformer_params(args, metadata)
             if not isinstance(transformer_params, dict):
-                return __internal__.handle_error(-101, \
-                    "Invalid return from getting transformer parameters from transformer class instance")
-            if 'code' in transformer_params:
-                if 'error' in transformer_params:
-                    error = transformer_params['error']
-                else:
-                    error = "Error returned from get_transformer_params with code: %s" % transformer_params['code']
-                return __internal__.handle_error(-101, error)
+                return __internal__.handle_error(-101,
+                                                 "Invalid return from getting transformer parameters from transformer class instance")
+
+            params_result = __internal__.check_params_result_error(transformer_params)
+            if params_result:
+                return params_result
         else:
-            logging.debug("Transformer class instance does not have get_transformer_params method")
+            logging.info("Transformer class instance does not have get_transformer_params method")
```

Comment:
I am not sure what best practices on logging are, but if I saw this I would have no idea what to do with this information. Are there any reasons to worry that this method is absent? Does it mean that it would be prudent to write one? Or if one is not needed, how does it help to know that it isn't there?

Reply:
See my comment below as well. The logging.info() is more for developers in that it needs to be explicitly specified. A person just running the container/code would not see this (since it's not a problem - not an Error, Warning, or Critical)

Reply:
but still, the questions I listed would be relevant to a developer ... it isn't immediately clear what to do if a particular method is missing

Reply:
The printed output looks like this: `INFO:root:Transformer class instance does not have get_transformer_params method`

If this were anything but just a piece of information, or debug-only detail, it should minimally be a warning.

To directly answer your question, the relevancy depends upon what problem a developer is seeing. Since it's informational, it's along the lines of "Hey, just letting you know this happened; do with it what you want". If the function is defined and getting called, they would put debugging information in the function.

It may become relevant if a developer is expecting their 'get_transformer_params()' function called but it isn't getting called - perhaps they have a typo in the name and are seeing unexpected behavior? I think the point of the message is that this call is optional (if it's not implemented, it's not a problem) and if they're having issues (indicated by enabling additional logging) it **may** be helpful for them to know what the code is doing.

If this response isn't addressing your concerns, lets have a chat about this